### PR TITLE
Upgraded to meshkit error

### LINKED ISSUE #3470: bitcask_test_profiles_persister.go

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Upgraded to meshkit error in modals package
- bitcask_test_profiles_persistor.go
- preference_map_persistor.go

This PR fixes #3470 #3484 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/bitcask_test_profiles_persister.go [hunk 1]
```diff
@@ -59,14 +54,13 @@ func NewBitCaskTestProfilesPersister(folderName string) (*BitCaskTestProfilesPer
...
-		return nil, errors.New("connection to DB does not exist")
+		return nil, ErrDBConnection
 	}
 
 RETRY:
 	locked, err := s.db.TryRLock()
 	if err != nil {
-		err = errors.Wrapf(err, "unable to obtain read lock from bitcask store")
-		logrus.Error(err)
+		ErrDBRLock(err)
```
Comment (kumarabd) [inline models/bitcask_test_profiles_persister.go:63]:
```suggestion
		logrus.Error(ErrDBRLock(err))
```

File: models/bitcask_test_profiles_persister.go [hunk 2]
```diff
@@ -128,14 +116,13 @@ RETRY:
...
```
Comment (kumarabd) [inline models/bitcask_test_profiles_persister.go:125]:
```suggestion
		logrus.Error(ErrDBRLock(err))
```

**Comment by sayantan1413** [comment]
@kumarabd  I think I forgot to return those errors should I update it?

File: models/bitcask_test_profiles_persister.go [hunk 3]
```diff
@@ -172,14 +153,13 @@ RETRY:
...
-		return errors.New("connection to DB does not exist")
+		return ErrDBConnection
 	}
 
 RETRY:
 	locked, err := s.db.TryRLock()
 	if err != nil {
-		err = errors.Wrapf(err, "unable to obtain read lock from bitcask store")
-		logrus.Error(err)
+		ErrDBRLock(err)
```
Comment (kumarabd) [inline models/bitcask_test_profiles_persister.go:162]:
```suggestion
		logrus.Error(ErrDBRLock(err))
```

File: models/bitcask_test_profiles_persister.go [hunk 4]
```diff
@@ -208,18 +184,17 @@ RETRY:
...
-		return errors.New("given result data is nil")
+		return ErrResultData()
 	}
 
 RETRY:
 	locked, err := s.db.TryLock()
 	if err != nil {
-		err = errors.Wrapf(err, "unable to obtain write lock from bitcask store")
-		logrus.Error(err)
+		ErrDBLock(err)
```
Comment (kumarabd) [inline models/bitcask_test_profiles_persister.go:197]:
```suggestion
		logrus.Error(ErrDBLock(err))
```

**Comment by kumarabd** [comment]
> @kumarabd I think I forgot to return those errors should I update it?

yes pls, I tried to add log for now.

**Comment by sayantan1413** [comment]
Okay am return the errors .

**Comment by kumarabd** [comment]
@sayantan1413 also pls remove the logrus from these packages when done.

**Comment by sayantan1413** [comment]
Hey @kumarabd how should I replace the `logrus.Debugf` and `logrus.Infof` ?

**Comment by kumarabd** [comment]
> Hey @kumarabd how should I replace the `logrus.Debugf` and `logrus.Infof` ?

Convert them to Debug and Info

**Comment by sayantan1413** [comment]
But @kumarabd I am getting an error doing so 😬

**Comment by sayantan1413** [comment]
Screenshot 2021-07-23 at 9 11 57 PM

**Comment by sayantan1413** [comment]
But @kumarabd I found `log.Debugf` is working fine should I use this?

File: models/preference_map_persister.go [hunk 5]
```diff
@@ -29,50 +28,49 @@ func (s *MapPreferencePersister) ReadFromPersister(userID string) (*Preference,
...
 
 // WriteToPersister persists session for the user
 func (s *MapPreferencePersister) WriteToPersister(userID string, data *Preference) error {
 	if s.db == nil {
-		return errors.New("connection to DB does not exist")
+		return ErrDBConnection
 	}
 
 	if userID == "" {
-		return errors.New("user ID is empty")
```
Comment (leecalcote) [inline models/preference_map_persister.go:62]: This text is helpful. Where is it now being captured?

**Comment by leecalcote** [comment]
Merge conflict.

File: models/preference_map_persister.go (hunk 5 above)
Comment (sayantan1413) [reply models/preference_map_persister.go:62]: Hey @leecalcote the variable `ErrUserID` has captured this text now. In error.go this `ErrUserID` is present. `	ErrUserID               = errors.New(ErrUserIDCode, errors.Alert, []string{"User ID is empty"}, []string{}, []string{}, []string{})
`

**Comment by sayantan1413** [comment]
Hey @leecalcote I think the merge conflict is solved now.

**Comment by sayantan1413** [comment]
Hey @dhruv0000 @kumarabd  I have used `log.Debugf`  imported from the `"fortio.org/fortio/log"` So is it right to be used here ?

**Comment by leecalcote** [comment]
> Hey @dhruv0000 @kumarabd I have used `log.Debugf` imported from the `"fortio.org/fortio/log"` So is it right to be used here ?

For logging, we should be using MeshKit's logger.

**Comment by sayantan1413** [comment]
Yeah but @leecalcote I am facing issue in that like I cannot change the `logrus.Debugf` to meshkit logging like . So if anyone can help me out here it would be really helpful to me that how should I change the `logrus.Debugf` in models package.

**Comment by leecalcote** [comment]
@sayantan1413 oh, I see. Yes, I think this will take an enhancement in MeshKit's logger. So long as we're not losing any logs in this PR and dealing only with errors, let's make sure that a separate enhancement issue is filed against MeshKit and constrain the focus of this PR to errors only.

**Comment by sayantan1413** [comment]
Okay @leecalcote so let me keep the `logrus.Debugf` as it for now for this PR and wait for the enhancement then I can make a separate PR to change it.

**Comment by sayantan1413** [comment]
Hey @leecalcote I have reverted it back to `logrus.Debugf` for now and only focus on the Error codes.

**Review by dhruv0000 — APPROVED** [review]

**Pull request merged**